{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0169dd0c-751d-4932-b0a8-c6875d16b312",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 69, in <module>\n",
      "    min_height = min(contactheight)\n",
      "ValueError: min() arg is an empty sequence\n",
      "\n",
      "Script ran successfully: Range: 16024um to 16038um (14um)\n",
      "Bins:\n",
      " - Low (l):          16024-16027um\n",
      " - Medium low (ml):  16027-16031um\n",
      " - Medium high (mh): 16031-16034um\n",
      " - High (h):         16034-16038um\n",
      "Mean: 16032.485um\n",
      "Median: 16032um\n",
      "Mode: 16036um\n",
      "Chips at median height: 81, A3, B4, A6, B6, C6, C7, B7, 47, 37, 27, 18, 78, 98, A9, 99, 59, 3D, 6F\n",
      "Chips at maximum height: 3A, 4A, 3B, 4C, 5C, 5D, 4D, 5E\n",
      "Chips at minimum height: 43\n",
      "+ Saved wafer map: /nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/Topography_map_NC0W14-07G5.pdf\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Script ran successfully: Range: 16022um to 16036um (14um)\n",
      "Bins:\n",
      " - Low (l):          16022-16025um\n",
      " - Medium low (ml):  16025-16029um\n",
      " - Medium high (mh): 16029-16032um\n",
      " - High (h):         16032-16036um\n",
      "Mean: 16030.235um\n",
      "Median: 16030um\n",
      "Mode: 16032um\n",
      "Chips at median height: 71, A3, B4, 26, B6, C6, C7, B7, 37, 27, 18, 78, 98, A8, A9, 99, 89, 79, 8B, 7B, 6B\n",
      "Chips at maximum height: D8, 29, 2A, 3A, 4A, 3B, 3C, 4D\n",
      "Chips at minimum height: 33\n",
      "+ Saved wafer map: /nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/Topography_map_NC0W14-05A4.pdf\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Script ran successfully: Range: 15996um to 16008um (12um)\n",
      "Bins:\n",
      " - Low (l):          15996-15999um\n",
      " - Medium low (ml):  15999-16002um\n",
      " - Medium high (mh): 16002-16005um\n",
      " - High (h):         16005-16008um\n",
      "Mean: 16002.706um\n",
      "Median: 16002um\n",
      "Mode: 16002um\n",
      "Chips at median height: 83, 84, B4, A5, 95, A6, B6, C6, B7, A7, 17, 28, 38, 48, 58, 68, 78, 88, 98, A8, B8, A9, 89, 69, 59, 49, 39, 29, 7A, 8A, 9A, 8B, 7B\n",
      "Chips at maximum height: 3C, 4C, 5C, AD, 4D, 3D, 4E\n",
      "Chips at minimum height: 43, 33, 35, 25\n",
      "+ Saved wafer map: /nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/Topography_map_NC0W14-03B6.pdf\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Script ran successfully: Range: 15982um to 16012um (30um)\n",
      "Bins:\n",
      " - Low (l):          15982-15989um\n",
      " - Medium low (ml):  15989-15997um\n",
      " - Medium high (mh): 15997-16004um\n",
      " - High (h):         16004-16012um\n",
      "Mean: 15997.265um\n",
      "Median: 15997um\n",
      "Mode: 15988um\n",
      "Calculated median not in data set. Chips at next highest height (15998um): ['18', '28', '38', '48', '98', 'A8', 'B8', '89', '79', '59']\n",
      "Chips at maximum height: 4D, 3D, 5E\n",
      "Chips at minimum height: 33\n",
      "+ Saved wafer map: /nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/Topography_map_NC0W14-06H2.pdf\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Script ran successfully: Range: 15970um to 16002um (32um)\n",
      "Bins:\n",
      " - Low (l):          15970-15978um\n",
      " - Medium low (ml):  15978-15986um\n",
      " - Medium high (mh): 15986-15994um\n",
      " - High (h):         15994-16002um\n",
      "Mean: 15986.382um\n",
      "Median: 15987um\n",
      "Mode: 15970um\n",
      "Calculated median not in data set. Chips at next highest height (15988um): ['42', '98', 'B8', '99', '79']\n",
      "Chips at maximum height: 3C, AD, 4D, 3D, 4E, 5E, 6E, 8E, 6F\n",
      "Chips at minimum height: 53, 43, 33, 24, 34, 44, 54, 64, 74, 84, 94, 85, 75, 65, 55, 45, 35, 25\n",
      "+ Saved wafer map: /nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/Topography_map_NC0W14-04B1.pdf\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n",
      "Error running script: Traceback (most recent call last):\n",
      "  File \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/topography_map.py\", line 26, in <module>\n",
      "    log_content = file.readlines()\n",
      "  File \"/usr/lib64/python3.9/codecs.py\", line 322, in decode\n",
      "    (result, consumed) = self._buffer_decode(data, self.errors, final)\n",
      "UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "import os\n",
    "\n",
    "root_path = os.path.expanduser(\"~/Desktop/log_files/\")\n",
    "for root, dirs, files in os.walk(root_path):\n",
    "    for file_name in files:\n",
    "        file_path = os.path.join(root, file_name)\n",
    "\n",
    "        # Command to execute your Python script (replace with your script path and any arguments)\n",
    "        script_path = \"/nfshome/natpenn/Desktop/CMS-Wafer-Testing/Topography-Maps/asymmetry_map.py\"\n",
    "        command = ['python3.9', script_path, file_path]\n",
    "\n",
    "        # Run the script as a subprocess\n",
    "        process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)\n",
    "\n",
    "        # Wait for the process to complete and capture output (optional)\n",
    "        stdout, stderr = process.communicate()\n",
    "\n",
    "        # Check if there was an error\n",
    "        if process.returncode != 0:\n",
    "            print(f\"Error running script: {stderr.decode('utf-8')}\")\n",
    "        else:\n",
    "            print(f\"Script ran successfully: {stdout.decode('utf-8')}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d00fa48-bd13-4110-a397-93ff5a5b0736",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
